src/shared/elf-util.c

```diff
@@ -266,6 +266,42 @@ static int thread_callback(Dwfl_Thread *thread, void *userdata) {
         return DWARF_CB_OK;
 }
 
+static void report_module_metadata(StackContext *c, const char *name, JsonVariant *metadata) {
+        assert(c);
+        assert(name);
+        assert(metadata);
+
+        if (!c->f)
+                return;
+
+        const char
+                *build_id = json_variant_string(json_variant_by_key(metadata, "buildId")),
+                *type = json_variant_string(json_variant_by_key(metadata, "type")),
+                *package = json_variant_string(json_variant_by_key(metadata, "name")),
+                *version = json_variant_string(json_variant_by_key(metadata, "version")),
+                *arch = json_variant_string(json_variant_by_key(metadata, "architecture"));
+
+        fprintf(c->f, "Module %s", name);
+        if (package) {
+                /* The version is only meaningful with a package name. Skip the
+                 * version/arch fields if package is unknown. Architecture is
+                 * meaningful even without version, so print that. */
+                fprintf(c->f, " from %s %s", type ?: "package", package);
+
+                char sep1 = '-', sep2 = '.';  /* rpm (and others?) */
```

Comment:
I'm not sure about trying to recreate the actual binary package file name? For example, the architecture will be wrong for debian, as it's `amd64` rather than `x86_64` - maybe just separate by spaces and avoid trying to get it right? It is not machine-readable output after all here

Reply:
My goal is to have a precise identifier of a package. In the rpm world, that just `n-e:v-r.a`, and various tools (rpm, dnf) expect this syntax. This isn't exactly the package name, in particular there is no extension, and `e:` is also skipped in the file name. But the two are very close.

If it's not possible to do something useful for debian in this fashion, we can use some other syntax… What'd be the most useful thing?

Reply:
you can build the same thing, but you'd need to map the binary arch to the package arch, eg x86_64 -> amd64, and so on